controllers/feeds/getFeedsV2.js

```diff
@@ -94,6 +94,47 @@ const isValidActivity = async (item, conditions) => {
   return (item.final_score || 0) >= threshold;
 };
 
+const getActorFromLatestReaction = (latestReaction) => {
+  let postActors = new Set();
+  latestReaction?.upvotes?.forEach((upvote) => postActors.add(upvote.user_id));
+  latestReaction?.downvotes?.forEach((downvote) => postActors.add(downvote.user_id));
+  latestReaction?.comments?.forEach((comments) => postActors.add(comments.user_id));
+  latestReaction?.comment?.forEach((comment) => postActors.add(comment.user_id));
+  return Array.from(postActors);
+};
+
+const setKarmaScore = (latestReaction, karmaScores) => {
+  return latestReaction.map((reaction) => {
+    const user = karmaScores.find((user) => user.user_id === reaction.user_id);
+    if (reaction.latest_children) {
+      reaction.latest_children = addKarmaScoreToLatestReaction(
+        reaction.latest_children,
+        karmaScores
+      );
+    }
+    return {
+      ...reaction,
+      karma_score: roundingKarmaScore(user?.karma_score || 0)
+    };
+  });
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `setKarmaScore` function, you are modifying the `reaction.latest_children` property directly. It's generally a bad practice to mutate function arguments as it can lead to unexpected side effects. Instead, consider creating a new object and returning that.

```diff
-    if (reaction.latest_children) {
-      reaction.latest_children = addKarmaScoreToLatestReaction(
-        reaction.latest_children,
-        karmaScores
-      );
-    }
+    const latest_children = reaction.latest_children 
+      ? addKarmaScoreToLatestReaction(reaction.latest_children, karmaScores)
+      : undefined;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->